cocotb/handle.py

```diff
@@ -651,14 +651,30 @@ def _set_value(self, value, call_sim):
                 The value to drive onto the simulator object.
 
         Raises:
-            TypeError: If target is not wide enough or has an unsupported type
-                 for value assignment.
+            TypeError: If target has an unsupported type for value assignment.
+
+            OverflowError: If int value is out of the range that can be represented
+                by the target. -2**(Nbits - 1) <= value <= 2**Nbits - 1
         """
         value, set_action = self._check_for_set_action(value)
 
-        if isinstance(value, int) and value < 0x7fffffff and len(self) <= 32:
-            call_sim(self._handle.set_signal_val_long, set_action, value)
-            return
+        if isinstance(value, int):
+            min_value = -2 ** (len(self) - 1)  # minimum negative value for 'signed' number
```

Comment:
```suggestion
            min_value = -(2 ** (len(self) - 1))  # minimum negative value for 'signed' number
```
negation is higher precedence than power, `-2 ** 3 = -8` in Python.

Reply:
That's not an interesting example, perhaps you meant `-2 ** 2 = -4`, which I think shows the opposite of what you claim

Reply:
Yep, not entirely sure what I was thinking. I feel like I got bit by something related to that before and am remembering wrong.

Reply:
Pretty sure the code works as intended